### PR TITLE
Duplicate functions are allowed as long as they have the same source

### PR DESCRIPTION
**Pull request by jystic**

### CODE REVIEW

**Comment by jystic** [comment]
We need this unbreak the build /cc @tranma

File: src/Icicle/Source/Checker/Function.hs [hunk 1]
```diff
@@ -37,13 +37,24 @@ checkFs :: Ord n
...
+    case List.lookup (snd name) env' of
+     Nothing
+      -> pure (env' <> [(snd name , (funtype, annotfun))])
+
+     Just (t, f)
+      | f'        <- reannotF (const ()) f
+      , annotfun' <- reannotF (const ()) annotfun
+      , t  == funtype
+      , f' == annotfun'
+      -> pure env'
```
Comment (tranma) [inline src/Icicle/Source/Checker/Function.hs]: will this consider alpha equivalence?

Comment (jystic) [inline src/Icicle/Source/Checker/Function.hs]: no I think it's just if the source is the same, it's just a `#pragma once`

**Comment by amosr** [comment]
:+1:

**Pull request merged**